client/src/app/pages/search/components/SearchMenu.tsx

```diff
@@ -0,0 +1,314 @@
+      }));
+
+      const transformedPackages: IEntity[] = packages.map((item) => ({
+        id: item.uuid,
+        title: item.decomposedPurl ? item.decomposedPurl?.name : item.purl,
+        description: item.decomposedPurl?.namespace,
+        navLink: `/packages/${item.uuid}`,
+        type: "Package",
+        typeColor: "cyan",
+      }));
+
+      const transformedSboms: IEntity[] = sboms.map((item) => ({
+        id: item.id,
+        title: item.name,
+        description: item.authors.join(", "),
+        navLink: `/sboms/${item.id}`,
+        type: "SBOM",
+        typeColor: "purple",
+      }));
+
+      const transformedVulnerabilities: IEntity[] = vulnerabilities.map(
+        (item) => ({
+          id: item.identifier,
+          title: item.identifier,
+          description: item.description?.substring(0, 75),
+          navLink: `/vulnerabilities/${item.identifier}`,
+          type: "CVE",
+          typeColor: "orange",
+        })
+      );
+
+      tmpArray.push(
+        ...transformedAdvisories,
+        ...transformedPackages,
+        ...transformedSboms,
+        ...transformedVulnerabilities
+      );
+      setEntityList(tmpArray);
+    }
+    // fetch on load
+    fetchAllEntities();
+  }, [advisories, packages, sboms, vulnerabilities]);
+  return {
+    list: entityList,
+    defaultValue:
+      sbomFilterState.filterValues[FILTER_TEXT_CATEGORY_KEY]?.[0] || "",
+  };
+}
+
+export interface ISearchMenu {
+  onChangeSearch: (searchValue: string | undefined) => void;
+}
+
+export const SearchMenu: React.FC<ISearchMenu> = ({ onChangeSearch }) => {
+  const { list: entityList, defaultValue } = useAllEntities();
+
+  const [searchValue, setSearchValue] = React.useState<string | undefined>(
+    defaultValue
+  );
+  const [autocompleteOptions, setAutocompleteOptions] = React.useState<
+    React.JSX.Element[]
+  >([]);
+  const [isAutocompleteOpen, setIsAutocompleteOpen] =
+    React.useState<boolean>(false);
+
+  const searchInputRef: React.RefObject<HTMLInputElement> = React.useRef(null);
+  const autocompleteRef: React.RefObject<HTMLInputElement> = React.useRef(null);
+
+  const onChangeSearchValue = (newValue: string) => {
+    if (
+      newValue !== "" &&
+      searchInputRef &&
+      searchInputRef.current &&
+      searchInputRef.current.contains(document.activeElement)
+    ) {
+      setIsAutocompleteOpen(true);
+
+      // When the value of the search input changes, build a list of no more than 10 autocomplete options.
+      // Options which start with the search input value are listed first, followed by options which contain
+      // the search input value.
+      let options: React.JSX.Element[] = entityList
+        .filter(
+          (option) =>
+            option.id.toLowerCase().startsWith(newValue.toLowerCase()) ||
+            option.title?.toLowerCase().startsWith(newValue.toLowerCase()) ||
+            option.description?.toLowerCase().startsWith(newValue.toLowerCase())
```

Comment:
I don't have anything in mind just yet but I wonder if the filtering logic could be done with a "contains" and not "startWith". Maybe in a second PR we can come up with some logic to appropriately filter by string. but we can leave this for later I think

Reply:
I moved the filter functionality into its own function, and I created a test for it. Then I created a story that shows how you can override the filter function, so we can show what it would look like if we change the logic. And there is an example with using `includes` instead of `startWith` but I can come back to it and make it better later on.